File: java-spring-virtual-threads/benchmark/src/main/java/com/example/benchmark/model/Square.java
package com.example.benchmark.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Square implements FigureInterface {
    private final int size;

    @JsonCreator
    Square(@JsonProperty("size") int size) {
        this.size = size;
    }

    @Override
    public String getType() {
        return "square";
    }

    @Override
    public double computeArea() {
        return Math.pow(size, 2);
    }
}

File: kotlin-ktor-jvm/src/main/kotlin/com/example/model/Square.kt
package com.example.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.math.pow

@Serializable
@SerialName("square")
class Square(
    private val size: Int
): Figure {
    override fun getType(): String {
        return "square";
    }

    override fun computeArea(): Double {
        return size.toDouble().pow(2.0);
    }
}